src/main/java/seedu/address/model/person/Person.java

```diff
@@ -104,15 +113,30 @@ public Person withPicture(Picture picture) {
     }
 
     public Person withDates(List<Event> dates) {
-        return new Person(name, phone, email, birthday, address, picture, tags, dates, meetings);
+        return new Person(name, phone, email, birthday, goal, address, picture, tags, dates, meetings);
+    }
+
+    public Person withGoal(Goal goal) {
+        return new Person(name, phone, email, birthday, goal, address, picture, tags, dates, meetings);
     }
 
     public List<Event> getMeetings() {
         return Collections.unmodifiableList(meetings);
     }
 
     public Person withMeetings(List<Event> meetings) {
-        return new Person(name, phone, email, birthday, address, picture, tags, dates, meetings);
+        return new Person(name, phone, email, birthday, goal, address, picture, tags, dates, meetings);
+    }
+
+    public LocalDate getGoalDeadline(LocalDate date) {
+        if (this.meetings.isEmpty()) {
+            // user has never met up with this person before
+            return DateUtil.ZERO_DAY;
+        }
+        return this.goal.getNext(this.meetings.stream()
```

Comment:
This runs `goal.getNext()` with the lastest meeting date? If it is, should probably write.
```java
Date latestMeetingDate = meetings.stream()
        .map(Event::getDate)
        .filter(x -> x.isBefore(date)
        .max(LocalDate::compareTo)
        .orElse(DateUtil.ZERO_DAY);
return goal.getNext(latestMeetingDate);
```